ios/BetterRailWidget/Live Activity/BetterRailLiveActivity.swift

```diff
@@ -44,12 +42,19 @@ struct BetterRailLiveActivity: Widget {
         } compactLeading: {
           CircularProgressView(vm: vm, content: .icon, tintColor: tintColor(context: context))
         } compactTrailing: {
-          Text("\(String(getMinutesLeft(targetDate: getStatusEndDate(context: context)))) min")
-            .foregroundColor(tintColor(context: context))
-            .accessibilityLabel(vm.status == .waitForTrain ? "time left depart" : "time left arrival")
-            .contentTransition(.numericText())
+          let endDate = getStatusEndDate(context: context)
+          
+          if (vm.isStale) {
+            Text(formatDateHour(endDate))
+              .foregroundColor(tintColor(context: context))
+          } else {
+            Text("\(String(getMinutesLeft(targetDate: endDate))) min")
+              .foregroundColor(tintColor(context: context))
+              .accessibilityLabel(vm.status == .waitForTrain ? "time left depart" : "time left arrival")
+              .contentTransition(.numericText())
+          }
         } minimal: {
-          CircularProgressView(vm: vm, content: .time, tintColor: tintColor(context: context))
+          CircularProgressView(vm: vm, content: vm.isStale ? .icon : .time, tintColor: tintColor(context: context))
```

Comment:
I prefer the stale icon logic to be in `CircularProgressView` with the rest of the conditions

Reply:
why? imo it's better to have this controlled by the consumer view

Reply:
There are many conditions where the time fallbacks to icon. This is one of them. Look in the `CircularProgressView` file.